spec/models/manageiq/providers/hawkular/middleware_manager/event_catcher/runner_spec.rb

```diff
@@ -80,4 +80,58 @@
       expect(hash[:full_data]).to be_an String
     end
   end
+
+  context "event_monitor" do
+    it "reset_event_monitor_handle" do
+      expect(subject.instance_variable_get('@event_monitor_handle')).to be_nil
+      subject.send(:reset_event_monitor_handle)
+    end
+
+    it "stop_event_monitor" do
+      expect(subject).to receive(:reset_event_monitor_handle)
+      expect(subject.instance_variable_get('@event_monitor_handle')).to be_nil
+      subject.send(:stop_event_monitor)
+    end
+
+    context "event_monitor_handle" do
+      it "event_monitor_handle not defined" do
+        VCR.use_cassette(described_class.name.underscore.to_s,
+                         :decode_compressed_response => true) do # , :record => :new_episodes) do
+          subject.instance_variable_set('@event_monitor_handle', nil)
+          subject.send(:event_monitor_handle)
+          expect(subject.instance_variable_get('@event_monitor_handle')).to be_an ManageIQ::Providers::Hawkular::MiddlewareManager::EventCatcher::Stream
+        end
+      end
+    end
+  end
+
+  it 'process_event' do
+    subject.instance_variable_set('@filtered_events', ['hawkular_event.critical', 'hawkular_datasource.ok'])
+    expect(subject.send(:blacklist?, "")).to be_falsey
+    expect(subject.send(:blacklist?, "hawkular_event.critical")).to be_truthy
+  end
+
+  it 'process_event' do
+    event = ::Hawkular::Alerts::Event.new({})
+    event.ctime = Time.now.to_i
+    event.text    = 'text message'
+    event.context = {'resource_path' => 'canonical_path', 'message' => 'context message'}
+    event.tags    = {'miq.event_type' => 'hawkular_event.critical', 'miq.resource_type' => 'MiddlewareServer'}
+    event_hash = subject.send(:event_to_hash, event, 1)
+    subject.instance_variable_set('@cfg', :ems_id => 1)
+    subject.instance_variable_set('@filtered_events', ['hawkular_datasource.ok'])
```

Comment:
I'm not in favor of including these three lines. The instance variables and the `event_to_hash` method are part of internal implementation of your subject `process_event`. By setting the instance variables you are forcing the test to pass even if future code changes set these variables to a wrong value. And by using `event_hash` in the expectation, you are not really testing/ensuring that `process_event` is interfacing correctly with `EmsEvent` (since `event_to_hash` is just part of internal implementation).

I know that unit tests are _white box_ tests. But this is only to ensure that all paths are tested by changing inputs and checking the outcomes. Not to be free to manipulate the objects :S (I'd do that only if really necessary).

Reply:
The idea was to have the arguments in the expectation of `add_queue`as hardcoded values, because we know how is the interface. So, any changes to the internal implementation of the `runner` that break the contract will be detected by the test.